pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -590,6 +602,56 @@ func validateDomain(domain string, validDomains []string) bool {
 	return false
 }
 
+func validateMachinePools(newObject *hivev1.ClusterDeployment, contextLogger *log.Entry) *admissionv1beta1.AdmissionResponse {
+
+	if reason, result := validateMachinePool(newObject, &newObject.Spec.ControlPlane, contextLogger); !result {
+		message := fmt.Sprintf("Control plane machine pool validation failure: %s", reason)
+		return &admissionv1beta1.AdmissionResponse{
+			Allowed: false,
+			Result: &metav1.Status{
+				Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+				Message: message,
+			},
+		}
+	}
+
+	for _, mp := range newObject.Spec.Compute {
+		if reason, result := validateMachinePool(newObject, &mp, contextLogger); !result {
+			message := fmt.Sprintf("Compute machine pool %s validation failure: %s", mp.Name, reason)
+			return &admissionv1beta1.AdmissionResponse{
+				Allowed: false,
+				Result: &metav1.Status{
+					Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+					Message: message,
+				},
+			}
+		}
+	}
+	return nil
+}
+
+func validateMachinePool(cd *hivev1.ClusterDeployment, mp *hivev1.MachinePool, contextLogger *log.Entry) (string, bool) {
+	if cd.Spec.PlatformSecrets.GCP != nil {
```

Comment:
This should check `cd.Spec.Platform.GCP` instead.

Reply:
done